tests/cbwd/watchdog.cc

```diff
@@ -79,6 +78,268 @@ TEST(WatchdogTest, BadConfig) {
+}
+
+TEST(WatchdogTest, Instance_config) {
+  std::string const& content{
+      "{\n"
+      "  \"centreonBroker\": {\n"
+      "   \"cbd\": [\n"
+      "      {\n"
+      "        \"executable\": \"" CENTREON_BROKER_WD_TEST
+      "tester\",\n"
+      "        \"name\": \"central-broker-master\",\n"
+      "        \"run\": true,\n"
+      "        \"reload\": true\n"
+      "      },\n"
+      "      {\n"
+      "        \"executable\": \"" CENTREON_BROKER_WD_TEST
+      "tester\",\n"
+      "        \"name\": \"central-rrd-master\",\n"
+      "        \"run\": true,\n"
+      "        \"reload\": true\n"
+      "      }\n"
+      "    ],\n"
+      "    \"log\": \"/tmp/watchdog.log\"\n"
+      "  }\n"
+      "}"};
+    create_conf("/tmp/instance-conf.json", content);
+    std::string result = com::centreon::broker::misc::exec(
+      "bin/cbwd /tmp/instance-conf.json"
+    );
+    ASSERT_EQ("[cbwd] [error] watchdog: Could not parse the configuration file"
+    " '/tmp/instance-conf.json': instance_config field not provided for cbd instance\n", result);
+}
+
+TEST(WatchdogTest, Run) {
+  std::string const& content{
+      "{\n"
+      "  \"centreonBroker\": {\n"
+      "   \"cbd\": [\n"
+      "      {\n"
+      "        \"executable\": \"" CENTREON_BROKER_WD_TEST
+      "tester\",\n"
+      "        \"name\": \"central-broker-master\",\n"
+      "        \"configuration_file\": \"Master\",\n"
+      "        \"reload\": true\n"
+      "      },\n"
+      "      {\n"
+      "        \"executable\": \"" CENTREON_BROKER_WD_TEST
+      "tester\",\n"
+      "        \"name\": \"central-rrd-master\",\n"
+      "        \"configuration_file\": \"Slave\",\n"
+      "        \"reload\": true\n"
+      "      }\n"
+      "    ],\n"
+      "    \"log\": \"/tmp/watchdog.log\"\n"
+      "  }\n"
+      "}"};
+    create_conf("/tmp/run-conf.json", content);
+    std::string result = com::centreon::broker::misc::exec(
+      "bin/cbwd /tmp/run-conf.json"
+    );
+    ASSERT_EQ("[cbwd] [error] watchdog: Could not parse the configuration file "
+    "'/tmp/run-conf.json': run field not provided for cbd instance\n", result);
+}
+
+TEST(WatchdogTest, Reload) {
+  std::string const& content{
+      "{\n"
+      "  \"centreonBroker\": {\n"
+      "   \"cbd\": [\n"
+      "      {\n"
+      "        \"executable\": \"" CENTREON_BROKER_WD_TEST
+      "tester\",\n"
+      "        \"name\": \"central-broker-master\",\n"
+      "        \"configuration_file\": \"Master\",\n"
+      "        \"run\": true\n"
+      "      },\n"
+      "      {\n"
+      "        \"executable\": \"" CENTREON_BROKER_WD_TEST
+      "tester\",\n"
+      "        \"name\": \"central-rrd-master\",\n"
+      "        \"configuration_file\": \"Slave\",\n"
+      "        \"run\": true\n"
+      "      }\n"
+      "    ],\n"
+      "    \"log\": \"/tmp/watchdog.log\"\n"
+      "  }\n"
+      "}"};
+    create_conf("/tmp/reload-conf.json", content);
```

Comment:
Here, you create a file /tmp/reload-conf.json
Could you remove it at the end of the test?
You can do the same for each test.